### ISSUE TITLE
[REQ][typescript-fetch] OperationId into middleware

### ISSUE THREAD

**Issue by HeikoStudt** [OpenAPITools/openapi-generator#13933]
### Is your feature request related to a problem? Please describe.
For internal logging purpose, we want to use the OperationID in our middleware.
Additionally, we imagine special treatment of custom middlewares, e.g. for responses to some login-operation in a logging or login-middleware

## Describe the solution you'd like
* in `apis.mustache`: add the operationId (nickname) as a parameter into the request context.
* In `runtime.mustache`: handle the new parameter in `request()` and the private `this.fetchApi()`.
* Add the operationId as an (optional) additional parameter into `middleware.pre`, `middleware.post` and `middleware.onError`

## Describe alternatives you've considered
* One could use the path instead. That one is rather long and, using path parameters, not the best idea anyways. To match some backend operation / logging, this does not work nicely as well The OperationID is the best ID of the operation we have.

## Additional context
* If the operationId is used optionally, there should not be any compatibility caveats
* It is an easy addition, we can provide a PR

**Comment by ChristianVaughn** [comment]
Did you ever try to modify the templates to do this yourself? I am currently trying to do the same thing with the OperationId and am struggling to get it implemented in api.mustache and runtime.mustache

**Comment by HeikoStudt** [comment]
Yes, indeed. While we did it originally on a somewhat older version of the templates (like summer 2022).

I have put our efforts into https://github.com/HeikoStudt/openapi-generator/tree/operationid-for-middleware. However, I did not compile / test / ... that version yet, so will not PR at this very moment.
